### PR TITLE
Deprecate vendor (3rd party) provisioners

### PR DESCRIPTION
**Pull request by pselle**
This notice follows the [recent announcement](https://discuss.hashicorp.com/t/notice-terraform-to-begin-deprecation-of-vendor-tool-specific-provisioners-starting-in-terraform-0-13-3/13997) that the Chef, Habitat, Puppet, and Salt-masterless provisioners are deprecated and will be removed in future version of Terraform.

### CODE REVIEW

File: website/docs/provisioners/chef.html.markdown [hunk 1]
```diff
@@ -12,8 +12,9 @@ The `chef` provisioner installs, configures and runs the Chef Client on a remote
 resource. The `chef` provisioner supports both `ssh` and `winrm` type
 [connections](/docs/provisioners/connection.html).
 
--> **Note:** Provisioners should only be used as a last resort. For most
-common situations there are better alternatives. For more information, see
+-> **Note:** This provisioner has been deprecated as of Terraform 0.13.3 and will be
+removed in a future version of Terraform. For most common situations there are better
+alternatives to using provisioners. For more information, see
 [the main Provisioners page](./).
```
Comment (pkolyvas) [inline website/docs/provisioners/chef.html.markdown:18]: I find myself thinking that linking to the migration guides you've penned might be helpful here? I'm not sure how they fit within the context of our official docs, but perhaps we'd be able to help practitioners who land here or show up here having inherited configurations which use these particular tools?

Comment (pselle) [reply website/docs/provisioners/chef.html.markdown:18]: Hmm, good point! Or maybe link to the discuss post? That could also work nicely for routing questions possibly.

Comment (pkolyvas) [reply website/docs/provisioners/chef.html.markdown:18]: I like that - I think I'm going to keep further doc changes out of scope for this PR, and create a docs-focused PR for further discussion.

**Review by pkolyvas — APPROVED** [review]
Thanks for tying this all together so nicely Pam. :)

**Review by alisdair — APPROVED** [review]

**Pull request merged**